pandas/core/indexing.py

```diff
@@ -2586,6 +2587,7 @@ def maybe_convert_indices(indices, n):
     IndexError : one of the converted indices either exceeded the number
         of elements (specified by `n`) OR was still negative.
     """
+    indices = deepcopy(indices)
```

Comment:
Can this be done down on line 2601?  Avoid a copy if at all possible

Reply:
Ok moving deepcopy to 2601 makes sense; will do. Can't see a way to avoid copying altogether.

Reply:
Should be able to just use `.copy()` instead of importing deepcopy, no?

Reply:
Yes, true. Will change to .copy()